File: libraries/rest-assured-http-scope/src/test/java/com/eyfout/example/http/ExampleStateScopeProvider.java
package com.eyfout.example.http;

import ht.eyfout.junit.jupiter.api.http.HttpAPIStateScopeProvider;
import ht.eyfout.junit.jupiter.api.http.HttpAPIWhenScope;

public class ExampleStateScopeProvider extends HttpAPIStateScopeProvider<ExampleGivenState> {
    @Override
    public ExampleGivenState givenState() {
        return new ExampleGivenState();
    }
}

File: examples/micronaut-app/src/test/kotlin/ht/eyfout/dmv/http/ExampleStateScopeProvider.kt
package ht.eyfout.dmv.http

import ht.eyfout.example.client.dmv.DMVClient
import ht.eyfout.junit.jupiter.gherkin.api.http.HttpAPIStateScopeProvider
import ht.eyfout.junit.jupiter.gherkin.api.http.HttpAPIThenScope
import ht.eyfout.junit.jupiter.gherkin.api.http.HttpAPIWhenScope
import jakarta.inject.Singleton

@Singleton
class ExampleStateScopeProvider(private val client: DMVClient) :
    HttpAPIStateScopeProvider<ClientGivenState, HttpAPIWhenScope, HttpAPIThenScope>() {
    override fun givenState(): ClientGivenState {
        return ClientGivenState(client = client)
    }
}